pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1705,25 +1687,11 @@ func (c *Cloud) GetVolumeLabels(volumeName KubernetesVolumeID) (map[string]strin
 	return labels, nil
 }
 
-// GetDiskPath implements Volumes.GetDiskPath
-func (c *Cloud) GetDiskPath(volumeName KubernetesVolumeID) (string, error) {
-	awsDisk, err := newAWSDisk(c, volumeName)
-	if err != nil {
-		return "", err
-	}
-	info, err := awsDisk.describeVolume()
-	if err != nil {
-		return "", err
-	}
-	if len(info.Attachments) == 0 {
-		return "", fmt.Errorf("No attachment to volume %s", volumeName)
-	}
-	return aws.StringValue(info.Attachments[0].Device), nil
-}
-
 // DiskIsAttached implements Volumes.DiskIsAttached
+// TODO: Remove this - we can inline it into DetachDisk (because it is only used there)
 func (c *Cloud) DiskIsAttached(diskName KubernetesVolumeID, nodeName types.NodeName) (bool, error) {
-	_, instance, err := c.getFullInstance(nodeName)
+	// NeverCache - but TODO remove this function
+	_, instance, err := c.getFullInstance(NeverCache, nodeName)
```

Comment:
This will, but hence the TODO to remove the function entirely.  Though I think your argument does apply to `DetachDisk`

The surprising thing is that a full refresh should be as quota-expensive as a single volume request.  That's not true with pagination, but DescribeVolumes actually returns all instances.

There are other concerns as well than just quota - we pull down a lot more data over the wire, for example.  But quota and correctness are much more important right now.

I'm not opposed to a single volume fetch, I just don't think we know we need it yet.

I could see us starting to do an LRU type fetching strategy, on the grounds that every time we fetch 1 volume we should actually fetch 50, but I suspect also that is simply a lot of complexity for not a lot of gain.